be/src/formats/orc/apache-orc/c++/src/Reader.cc

```diff
@@ -384,40 +384,20 @@ void RowReaderImpl::loadStripeIndex() {
 
     // obtain row indexes for selected columns
     uint64_t offset = currentStripeInfo.offset();
-
-    static const uint64_t MAX_ONE_READ_ROW_INDEX_SIZE = 1024 * 1024;
     uint64_t rowIndexSize = currentStripeInfo.indexlength();
-    uint64_t startOffset = offset;
-    // rowIndexBufferData is allocated in rowIndexInputStream
-    std::unique_ptr<SeekableFileInputStream> rowIndexInputStream = nullptr;
-    const char* rowIndexBufferData = nullptr;
-    if (rowIndexSize < MAX_ONE_READ_ROW_INDEX_SIZE) {
-        rowIndexInputStream = std::unique_ptr<SeekableFileInputStream>(new SeekableFileInputStream(
-                contents->stream.get(), startOffset, rowIndexSize, *contents->pool, rowIndexSize));
-        int size = 0;
-        const void* data = nullptr;
-        if (!rowIndexInputStream->Next(&data, &size) || static_cast<uint64_t>(size) != rowIndexSize) {
-            throw ParseError("Failed to read the row index data");
-        }
-        rowIndexBufferData = static_cast<const char*>(data);
-    }
+    contents->stream->prepare(InputStream::PrepareEvent::READ_ROW_GROUP_INDEX, offset, rowIndexSize);
```

Comment:
Rollback to old version of reading row group index data.  Because we are going to handle small read in a unified framework.